.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,25 @@
+Thank you for submitting the pull request to the Apache Ignite.
+
+In order to streamline the review of the contribution 
+we ask you to ensure the following steps have been taken:
+
+### The Contribution Checklist
+- [ ] The JIRA ticket has been associated with the pull request
+- [ ] The JIRA ticket has the _Patch Available_ state
+- [ ] The pull request briefly describes changes that have been made
+- [ ] The pull request title has the following pattern 
+e.g. `IGNITE-12407: Java thin client. Cluster API support`
```

Comment:
I think we should provide a detailed commit message guide on Ignite Wiki, and add a link here.
Lots of good guidance here: https://chris.beams.io/posts/git-commit/

PR title and description should be used for the final squashed commit.

Reply:
I've added `tread PR title as final commit message` to the checklist. 
There is a link to the `How to Contribute` wiki page that exists below. We can mention other commit message requirements there.